frontend/app/views/spree/products/_thumbnails.html.erb

```diff
@@ -3,15 +3,17 @@
   <ul id="product-thumbnails" class="thumbnails list-inline" data-hook>
     <% @product.images.each do |i| %>
       <li class='tmb-all tmb-<%= i.viewable.id %>'>
-        <%= link_to(image_tag(main_app.url_for(i.url(:mini)), class: "thumbnail"), main_app.url_for(i.url(:product))) %>
+        <% img_tag = i.alt.present? ? image_tag(main_app.url_for(i.url(:mini)), class: "thumbnail", alt: i.alt) : image_tag(main_app.url_for(i.url(:mini)), class: "thumbnail", alt: image_alt(main_app.url_for(i.url(:mini)))) %>
```

Comment:
@avneetmalhotra Can you create a helper inside `frontend_helper.rb` to generate alt attribute? This will make code cleaner and remove the duplication. We also want to avoid putting logic inside views.

Something like this: 

```ruby
def set_image_alt(image, size)
  image.alt.present? ? image.alt : image_alt(main_app.url_for(image.url(size)))
end
```

And inside the view we will have:

```erb
<% img_tag = image_tag(
  main_app.url_for(i.url(:mini)),
  class: "thumbnail",
  alt: set_image_alt(i, :mini)
) %>
<%= link_to(img_tag, main_app.url_for(i.url(:product))) %>
```

Reply:
@devilcoders Yeah, will do.